src/metabase/notification/send.clj

```diff
@@ -343,18 +354,10 @@
                             (catch InterruptedException _
                               (log/warn "Interrupted while waiting for notification executor to terminate"))))))
     (dotimes [_ pool-size]
-      (.submit ^ExecutorService executor
-               ^Callable (fn []
-                           (while (not (Thread/interrupted))
-                             (try
-                               (let [notification (take-notification! queue)]
-                                 (send-notification-sync! notification))
-                               (catch InterruptedException _
-                                 (log/info "Notification worker interrupted, shutting down")
-                                 (throw (InterruptedException.)))
-                               (catch Exception e
-                                 (log/error e "Error in notification worker")))))))
+      (start-worker!))
     (fn [notification]
+      ;; ensure there is always a worker to handle the notification
+      (start-worker!)
```

Comment:
this is fine to call whenever we submit a notification because the pool is a fixed pool